ext/curl/multi.c

```diff
@@ -174,6 +174,29 @@ PHP_FUNCTION(curl_multi_remove_handle)
 }
 /* }}} */
 
+PHP_FUNCTION(curl_multi_get_handles)
+{
+	zval      *z_mh;
+	php_curlm *mh;
+
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+		Z_PARAM_OBJECT_OF_CLASS(z_mh, curl_multi_ce)
+	ZEND_PARSE_PARAMETERS_END();
+
+	mh = Z_CURL_MULTI_P(z_mh);
+
+	array_init(return_value);
+	zend_llist_position pos;
+	zval	*pz_ch;
```

Comment:
I've stolen the logic as-is from `curl_multi_exec` and did not adjust the code style to keep the file internally consistent. This also applies to the other remark.

There would be more to clean up here, for example the removal of the useless `(zval *)` cast, because a `void*` casts implicitly to anything.

Perhaps it would even make sense to migrate from `zend_llist` to `HashTable`.

Reply:
Sure, let's move this to a follow-up PR then.